src/pages/goals/[goal_id].tsx

```diff
@@ -0,0 +1,205 @@
+  return (
+    <div
+      className={`${$GoalPage.benchmarksTab}
+        ${value === currentlySelected ? $GoalPage.benchmarksTabSelected : ""} `}
+      onClick={(e) => {
+        handleSelect(e);
+      }}
+    >
+      {text}
+    </div>
+  );
+};
+
+const ViewGoalPage = () => {
+  const utils = trpc.useContext();
+
+  const router = useRouter();
+  const { goal_id } = router.query;
+
+  const [editGoal, setEditGoal] = useState(false);
+  const [editGoalInput, setEditGoalInput] = useState("");
+
+  const [activeTab, setActiveTab] = useState<selectionValue>(
+    selectionValue.ACTIVE
+  );
+
+  const { data: goal, isLoading } = trpc.iep.getGoal.useQuery({
+    goal_id: goal_id as string,
+  });
+
+  const showEditGoal = () => {
+    setEditGoal(true);
+    setEditGoalInput(goal?.description || "");
+  };
+
+  // TODO: modify callbacks for toast notification
+  const editMutation = trpc.iep.editGoal.useMutation({
+    onSuccess: () => utils.iep.getGoal.invalidate(),
+    onError: (err) => console.log({ err }),
+  });
+
+  const submitEditGoal = () => {
+    editMutation.mutate({
+      goal_id: goal_id as string,
+      description: editGoalInput,
+    });
+    setEditGoal(false);
+    setEditGoalInput("");
+  };
+
+  const cancelEditGoal = () => {
+    setEditGoal(false);
+    setEditGoalInput("");
+  };
+
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <Stack
+      spacing={2}
+      sx={{
+        display: "flex",
+        flexDirection: "column",
+        alignItems: "center",
+        width: "100%",
+        maxWidth: "1200px",
+        margin: "auto",
+      }}
+    >
+      <Grid container justifyContent="space-between">
+        <Grid item>
+          <h2>Goal</h2>
+        </Grid>
+      </Grid>
+
+      {/* Goal Description */}
+      <Container className={$GoalPage.goalDescriptionContainer}>
+        {!editGoal && (
+          <>
+            <p>{goal?.description}</p>
+            <button
+              className={$button.default}
+              onClick={showEditGoal}
+              style={{
+                margin: "auto",
+                marginTop: "1rem",
+                marginBottom: "1rem",
+              }}
+            >
+              Edit Goal
+            </button>
+          </>
+        )}
+        {editGoal && (
+          <form style={{ marginTop: "1rem" }} onSubmit={submitEditGoal}>
+            <textarea
+              value={editGoalInput}
+              name="description"
+              onChange={(e) => {
+                setEditGoalInput(e.target.value);
+              }}
+              className={$GoalPage.editGoalFormTextArea}
+            />
+            <Grid
+              container
+              justifyContent="space-around"
+              marginY={2}
+              md={6}
+              marginX="auto"
+            >
+              <Grid item>
+                <button className={$button.secondary} onClick={cancelEditGoal}>
+                  Cancel Changes
```

Comment:
We may be required to update the text to simply "Cancel" and "Save" for these, per the Design System. Feel free to bring it up in a meeting or mention it to the designers if you think it is better for users this way.

Reply:
I removed the word "Changes" from both buttons. As it is, there's not much to indicate that a goal is being edited aside from the textarea. The designers might want to add some text to the design that just says "Editing Goal" or something like that to make it more clear, but I personally don't feel strongly enough to get involved in that discussion.